tests/cnf/ran/ztp/tests/ztp-argocd-policies-app.go

```diff
@@ -0,0 +1,342 @@
+				Expect(err).ToNot(HaveOccurred(), "Failed to check if ZTP version is in range")
+
+				if !versionInRange {
+					Skip("This test requires a ZTP version of at least 4.14")
+				}
+			})
+
+			// 62260 - Same source CR file name
+			It("should validate the same source cr file name", reportxml.ID("62260"), func() {
+				By("updating Argo CD policies app")
+				exists, err := helper.UpdateArgoCdAppGitPath(
+					tsparams.ArgoCdPoliciesAppName, tsparams.ZtpTestPathCustomSourceReplaceExisting, true)
+				if !exists {
+					Skip(err.Error())
+				}
+
+				Expect(err).ToNot(HaveOccurred(), "Failed to update Argo CD git path")
+
+				By("waiting for policy to exist")
+				policy, err := helper.WaitForPolicyToExist(
+					HubAPIClient, tsparams.CustomSourceCrPolicyName, tsparams.TestNamespace, tsparams.ArgoCdChangeTimeout)
+				Expect(err).ToNot(HaveOccurred(), "Failed to wait for policy to exist")
+
+				By("waiting for the policy to be Compliant")
+				err = policy.WaitUntilComplianceState(policiesv1.Compliant, tsparams.ArgoCdChangeTimeout)
+				Expect(err).ToNot(HaveOccurred(), "Failed to wait for policy to be Compliant")
+
+				By("checking the custom namespace exists")
+				_, err = namespace.Pull(Spoke1APIClient, tsparams.CustomSourceCrName)
+				Expect(err).ToNot(HaveOccurred(), "Failed to pull namespace that should exist")
+			})
+
+			// 63516 - Reference non-existence source CR yaml file
+			It("verifies a proper error is returned in ArgoCD app when a non-existent"+
+				" source-cr is used in PGT", reportxml.ID("63516"), func() {
+				By("updating Argo CD policies app")
+				exists, err := helper.UpdateArgoCdAppGitPath(
+					tsparams.ArgoCdPoliciesAppName, tsparams.ZtpTestPathCustomSourceNoCrFile, false)
+				if !exists {
+					Skip(err.Error())
+				}
+
+				Expect(err).ToNot(HaveOccurred(), "Failed to update Argo CD git path")
+
+				By("checking the Argo CD conditions for the expected error")
+				err = helper.WaitForConditionInArgoCdApp(
+					HubAPIClient,
+					tsparams.ArgoCdPoliciesAppName,
+					ranparam.OpenshiftGitOpsNamespace,
+					"test/NoCustomCr.yaml is not found",
+					tsparams.ArgoCdChangeTimeout)
+				Expect(err).ToNot(HaveOccurred(), "Failed to check Argo CD conditions for expected error")
+			})
+
+			// 64407 - Verify source CR search path implementation
+			It("should validate the source CR search path implemented on a per source file name", reportxml.ID("64407"), func() {
+				By("checking service account does not exist on spoke")
+				_, err := serviceaccount.Pull(
+					Spoke1APIClient, tsparams.CustomSourceCrName, tsparams.CustomSourceTestNamespace)
+				Expect(err).To(HaveOccurred(), "Service account already exists before test")
+
+				By("checking storage class does not exist on spoke")
+				_, err = storage.PullClass(Spoke1APIClient, tsparams.CustomSourceStorageClass)
+				Expect(err).To(HaveOccurred(), "Storage class already exists before test")
+
+				By("updating Argo CD policies app")
```

Comment:
@yliu127 updating the policies app is what ends up creating the service account and storage class so if we want to verify they don't exist before the test this can't be the first step

Reply:
I was thinking of the skipping part to make it a bit more efficient. But I think this is very minor and not worth changing it to make it different than other test cases :)